spec/jobs/pds_cascading_search_job_spec.rb

```diff
@@ -0,0 +1,270 @@
+        
+        expect(patient_changeset.search_results.count).to eq(1)
+        expect(patient_changeset.search_results.first).to include(
+          "step" => "no_fuzzy_with_history",
+          "result" => "one_match",
+          "nhs_number" => "9449306168"
+        )
+      end
+    end
+
+    context "when no match found, cascading through steps" do
+      before do
+        allow(PDS::Patient).to receive(:search).and_return(nil)
+      end
+
+      it "enqueues next search step on no match" do
+        expect {
+          described_class.perform_now(
+            searchable_id: patient_changeset.id,
+            searchable_type: "PatientChangeset",
+            step_name: :no_fuzzy_with_history
+          )
+        }.to have_enqueued_job(described_class)
+          .with(
+            searchable_id: patient_changeset.id,
+            searchable_type: "PatientChangeset",
+            step_name: :no_fuzzy_with_wildcard_postcode
+          )
+      end
+    end
+
+    context "when no postcode provided" do
+      let(:patient_changeset) do
+        create(
+          :patient_changeset,
+          import: import,
+          pending_changes: {
+            "child" => {
+              "given_name" => "Charlie",
+              "family_name" => "Brown",
+              "date_of_birth" => "2010-01-01",
+              "address_postcode" => nil
+            },
+            "search_results" => []
+          }
+        )
+      end
+
+      it "records no_postcode result and enqueues ProcessPatientChangesetJob" do
+        expect {
+          described_class.perform_now(
+            searchable_id: patient_changeset.id,
+            searchable_type: "PatientChangeset"
+          )
+        }.to have_enqueued_job(ProcessPatientChangesetJob)
+          .with(patient_changeset)
+
+        patient_changeset.reload
+        
+        expect(patient_changeset.search_results.first).to include(
+          "step" => "no_fuzzy_with_history",
+          "result" => "no_postcode"
+        )
+      end
+    end
+
+    context "when name too short for wildcard search" do
+      let(:patient_changeset) do
+        create(
+          :patient_changeset,
+          import: import,
+          pending_changes: {
+            "child" => {
+              "given_name" => "Ed",
+              "family_name" => "Li",
+              "date_of_birth" => "2010-01-01",
+              "address_postcode" => "SW1A 1AA"
+            },
+            "search_results" => []
+          }
+        )
+      end
+
+      before do
+        allow(PDS::Patient).to receive(:search).and_return(nil)
+      end
+
+      it "skips wildcard name steps and completes search" do
+        described_class.perform_now(
+          searchable_id: patient_changeset.id,
+          searchable_type: "PatientChangeset"
+        )
+        
+        while enqueued_jobs.any? { |job| job["job_class"] == described_class.name }
+          perform_enqueued_jobs(only: described_class)
+        end
```

Comment:
Since we have code like this in a few places now I'm wondering if it's worth extracting it in to a common method that we can use in various parts of the tests.